### PR TITLE
Spotctl

### PR DESCRIPTION
**Pull request by ntfrnzn**
feat: support integration with spotctl

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -100,8 +100,8 @@ var _ = BeforeSuite(func(done Done) {
 	Expect(err).ToNot(HaveOccurred())
 
 	// there should be no preexisting helm release
-	helm := controller.getInstaller(controller.getClient, log)
-	rel, err := helm.Get(install.GetReleaseName(string(v1alpha1.SparkHistoryChartName)))
+	helm := controller.getInstaller(Wave, controller.getClient, log)
```
Comment (ntfrnzn) [inline controllers/suite_test.go:103]: This change sets the helm release prefix, and allows spotctl to use something other than "wave-"

File: controllers/wavecomponent_controller.go [hunk 2]
```diff
@@ -286,36 +287,53 @@ func (r *WaveComponentReconciler) reconcilePresent(ctx context.Context, req ctrl
...
 	if err != nil {
 		log.Error(err, "cannot get current conditions")
 		return ctrl.Result{}, err
 	}
 	if conditions != nil {
-		deepCopy := comp.DeepCopy()
-		changed := false
 		for _, c := range conditions {
-			changed = changed || SetWaveComponentCondition(&(deepCopy.Status), *c)
+			up := SetWaveComponentCondition(&(deepCopy.Status), *c)
```
Comment (ntfrnzn) [inline controllers/wavecomponent_controller.go:302]: this is a bug-fix: must make sure that SetWaveComponentCondition even if changed  is false

File: controllers/wavecomponent_controller.go [hunk 3]
```diff
@@ -286,36 +287,53 @@ func (r *WaveComponentReconciler) reconcilePresent(ctx context.Context, req ctrl
...
+		err := r.Client.Patch(ctx, deepCopy, client.MergeFrom(comp))
+		if err != nil {
+			log.Error(err, "patch error")
+			return ctrl.Result{}, err
 		}
+	}
 
+	condition := GetCurrentComponentCondition(deepCopy.Status)
+	requeue := true
+	if condition.Type == v1alpha1.WaveComponentAvailable && condition.Status == v1.ConditionTrue {
```
Comment (ntfrnzn) [inline controllers/wavecomponent_controller.go:325]: trigger a requeue if the status is not "available"

File: controllers/wavecomponent_controller.go [hunk 4]
```diff
@@ -479,7 +497,8 @@ func (r *WaveComponentReconciler) GetCurrentConditions(comp *v1alpha1.WaveCompon
 	case v1alpha1.SparkHistoryChartName:
 		return components.GetSparkHistoryConditions(r.Client, r.Log)
 	case v1alpha1.EnterpriseGatewayChartName:
-		return []*v1alpha1.WaveComponentCondition{conditionOK}, nil
+		restconfig, _ := r.getClient.ToRESTConfig()
+		return components.GetEnterpriseGatewayConditions(restconfig, r.Client, r.Log)
```
Comment (ntfrnzn) [inline controllers/wavecomponent_controller.go:501]: bug-fix: need to call the method!

File: internal/components/enterprisegateway.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+	if deployment.Status.AvailableReplicas == 0 {
+		conditions = append(conditions,
+			NewWaveComponentCondition(v1alpha1.WaveComponentAvailable, v1.ConditionFalse, "PodsUnavailable", "no pods are available"))
+		return conditions, nil
+	}
+	_, err = getEnterpriseGatewayEndpoint(ctx, client, log)
+	if err != nil {
+		conditions = append(conditions,
+			NewWaveComponentCondition(v1alpha1.WaveComponentDegraded, v1.ConditionTrue,
```
Comment (ntfrnzn) [inline internal/components/enterprisegateway.go:56]: A first use of "degraded", because the nginx-ingress has not populated the endpoint.

**Review by thorsteinnth — APPROVED** [review]

**Pull request merged**